### PR TITLE
 student pr

### PR DESCRIPTION
**Pull request by m83city**

### CODE REVIEW

File: students/src/main/java/com/example/students/controller/StudentController.java [hunk 1]
```diff
@@ -14,5 +14,37 @@
 @RestController
 @RequiredArgsConstructor
 public class StudentController {
+    private final StudentService studentService;
 
+    @GetMapping("/student/{id}/name/full")
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/controller/StudentController.java:19]: it means that you're getting from name resource some "full" field

File: students/src/main/java/com/example/students/controller/StudentController.java [hunk 2]
```diff
@@ -14,5 +14,37 @@
...
+    public StudentFullName getFullName (@PathVariable Integer id) {
+        if(isNull(id)){
+            throw new RuntimeException("Student Id is null");
+        }
+        return studentService.getStudentFullName(id);
+    }
+    @GetMapping("/student")
+    public List<StudentEntity> getAll() {
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/controller/StudentController.java:27]: why do you use Entity in service layer

File: students/src/main/java/com/example/students/controller/StudentController.java [hunk 3]
```diff
@@ -14,5 +14,37 @@
...
+            return ResponseEntity
+                    .status(HttpStatus.BAD_REQUEST)
+                    .eTag("HELLO WOLRD") // write body
+                    .build();
+        }
+        StudentDTO response = studentService.createStudent(studentDTO);
+        return ResponseEntity.ok(response);
+    }
+    @PutMapping("/student/{id}")
+    public StudentDTO updateById (@PathVariable Integer id, @RequestBody StudentDTO studentDTO){
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/controller/StudentController.java:43]: why do you use DTO in service layer. think should be Student

File: students/src/main/java/com/example/students/mapper/StudentMapper.java [hunk 4]
```diff
@@ -4,5 +4,24 @@
 import com.example.students.model.StudentEntity;
 public class StudentMapper {
 
+    public StudentEntity entityMapper(StudentDTO studentDTO){
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/mapper/StudentMapper.java:7]: make all these methods as static

File: students/src/main/java/com/example/students/service/StudentService.java [hunk 5]
```diff
@@ -15,5 +15,51 @@
...
 @RequiredArgsConstructor
 public class StudentService {
+    private final StudentRepository studentRepository;
+    private final StudentClient client;
+    StudentMapper studentMapper = new StudentMapper();
+    public StudentFullName getStudentFullName (Integer id) {
+        RestTemplate restTemplate = new RestTemplate();
+        StudentFullName fullName = new StudentFullName();
 
+        String resourceURL = String.format("http://name:8085/name/%d", id);// service: first-name
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/service/StudentService.java:25]: extract this string into application.yml

File: students/src/main/java/com/example/students/mapper/StudentMapper.java (hunk 4 above)
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/mapper/StudentMapper.java:7]: name it like "toStudentEntity" or "asStudentEntity"

File: students/src/main/java/com/example/students/mapper/StudentMapper.java [hunk 6]
```diff
@@ -4,5 +4,24 @@
...
+        return StudentEntity.builder()
+                .age(studentDTO.getAge())
+                .course(studentDTO.getCourse())
+                .firstName(studentDTO.getFirstName())
+                .lastName(studentDTO.getLastName())
+                .secondName(studentDTO.getSecondName())
+                .build();
+    }
+    public StudentDTO dtoMapper(StudentEntity studentEntity){
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/mapper/StudentMapper.java:16]: the same

File: students/src/main/java/com/example/students/service/StudentService.java [hunk 7]
```diff
@@ -15,5 +15,51 @@
...
+    public List<StudentEntity> getAllStudents(){
+        return studentRepository.findAll();
+    }
+    public Optional<StudentEntity> getStudentById(Integer id){
+        return studentRepository.findById(id);
+    }
+    public Integer deleteStudentById (Integer id){
+        studentRepository.deleteById(id);
+        return  id;
+    }
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/service/StudentService.java:54]: add everywhere extraspaces

File: students/src/main/java/com/example/students/service/StudentService.java [hunk 8]
```diff
@@ -15,5 +15,51 @@
...
+    public StudentDTO updateStudent (Integer id, StudentDTO studentDTO){
+            StudentEntity studentEntity = studentRepository.getReferenceById(id);
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/service/StudentService.java:56]: maybe should be like findById or getById

File: students/src/main/java/com/example/students/service/StudentService.java [hunk 9]
```diff
@@ -15,5 +15,51 @@
...
+            studentEntity.setAge(studentDTO.getAge());
+            studentEntity.setCourse(studentDTO.getCourse());
+            studentEntity.setFirstName(studentDTO.getFirstName());
+            studentEntity.setSecondName(studentDTO.getSecondName());
```
Comment (vlad-vinskevitch) [inline students/src/main/java/com/example/students/service/StudentService.java:60]: maybe need to extract into mapper

File: students/src/main/java/com/example/students/controller/StudentController.java (hunk 1 above)
Comment (m83city) [reply students/src/main/java/com/example/students/controller/StudentController.java:19]: yes i try to use this pattern:
```
| -- name :         |--first
                    |--second
                    |--last
                    |--full
```

Comment (m83city) [reply students/src/main/java/com/example/students/controller/StudentController.java:19]: updated

File: students/src/main/java/com/example/students/service/StudentService.java (hunk 5 above)
Comment (m83city) [reply students/src/main/java/com/example/students/service/StudentService.java:25]: neeed meet

File: students/src/main/java/com/example/students/service/StudentService.java (hunk 8 above)
Comment (m83city) [reply students/src/main/java/com/example/students/service/StudentService.java:56]: ***getReferenceById(id)*** is an internal method SpringBoot. And this method is updated

Comment (m83city) [reply students/src/main/java/com/example/students/service/StudentService.java:56]: `getById()` on documentation marked as deprecated and replaced `getReferenceById()`

**Pull request closed without merging**